### PR TITLE
IcingaDB: don't always set icinga:config:*command:{argument,envvar}#value

### PR DESCRIPTION
**Pull request by Al2Klimov**

### CODE REVIEW

File: lib/icingadb/icingadb-objects.cpp [hunk 1]
```diff
@@ -698,7 +698,14 @@ void IcingaDB::InsertObjectDependencies(const ConfigObject::Ptr& object, const S
 					values = new Dictionary({{"value", kv.second}});
 				}
 
-				values->Set("value", JsonEncode(values->Get("value")));
+				{
+					Value value;
+
+					if (values->Get("value", &value)) {
```
Comment (dnsmichi) [inline lib/icingadb/icingadb-objects.cpp]: Shouldn't the logic be inverted?

Or rather, use `if (!values->Contains("value"))` which is less resource consuming.

Comment (Al2Klimov) [reply lib/icingadb/icingadb-objects.cpp]: 1. What for?
2. It consumes even more resources.

Comment (dnsmichi) [reply lib/icingadb/icingadb-objects.cpp]: Nevermind. The code is barely readable.

Actually you're performing a "replace into", but only if there's a value. Maybe add a comment over the block to let others understand it.

Comment (N-o-X) [reply lib/icingadb/icingadb-objects.cpp]: Yeah, I think a small code comment would be nice in this case.

**Pull request merged**